src/Take.Blip.Builder/Strategies/CrmContext.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Builder.Models;
+using Take.Blip.Builder.Utils;
+
+namespace Take.Blip.Builder.Strategies
+{
+    public class CrmContext : ICrmContext
+    {
+        private ICrmProcessor _crmProcessor;
+        public async Task ExecuteAsync(IContext context, CrmSettings settings, CrmActionType actionType, CancellationToken cancellationToken)
+        {
+            if (actionType.Equals(CrmActionType.GetLead))
+            {
+                await _crmProcessor.GetLeadAsync(context, settings, cancellationToken);
+            }
+            else if (actionType.Equals(CrmActionType.CreateLead))
+            {
+                await _crmProcessor.RegisterLeadAsync(context, settings, cancellationToken);
+            }
+        }
+
+        public void SetCrm(ICrmProcessor crmProcessor)
+        {
+            _crmProcessor = crmProcessor;
```

Comment:
Could there be any other side-effects for mutating this context? Maybe if it's register as singleton.

Can the `crmProcesso` be passed as a param where it'll be used?